setup-babel.js

```diff
@@ -0,0 +1,68 @@
+/* eslint-env node */
+'use strict';
+
+const path = require('path');
+const VersionChecker = require('ember-cli-version-checker');
+
+function requireTransform(transformName) {
+  let plugin = require(transformName);
+
+  plugin = plugin.__esModule ? plugin.default : plugin;
+
+  // adding `baseDir` ensures that broccoli-babel-transpiler does not
+  // issue a warning and opt out of caching
+  const pluginPath = require.resolve(`${transformName}/package`);
+  const pluginBaseDir = path.dirname(pluginPath);
+  plugin.baseDir = () => pluginBaseDir;
+
+  return plugin;
+}
+
+function hasPlugin(plugins, name) {
+  for (const maybePlugin of plugins) {
+    const plugin = Array.isArray(maybePlugin) ? maybePlugin[0] : maybePlugin;
+    const pluginName = typeof plugin === 'string' ? plugin : plugin.name;
+
+    if (pluginName === name) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+module.exports = function setupBabel(parent) {
+  // Create parent options, if they do nox exist
+  const parentOptions = (parent.options = parent.options || {});
+
+  const checker = new VersionChecker(parent).for('ember-cli-babel', 'npm');
```

Comment:
This originally accessed `this.parent` instead of the `parent` argument that `included(parent)` receives. Can this change cause problems? I did not discover any so far.